File: Practice/15/Python/ex15.py
import random

u = 1
print ("Здравствуйте! Вас приветствует игра \"Угадай число\". Компьютер загадает случайное число от 0 до 100.\nВаша задача угадать это число. На всё про всё вам даётся 5 попыток. Желаем вам удачи!")
while u:
    print("Поехали!")
    max_user_try = 5
    user_try = 1
    n = random.randint(0, 100)
    while user_try <= max_user_try:
        user_change = int(input())
        if user_change == n:
            print ("Поздравляю! Вы угадали")
            break
        elif user_try == max_user_try:
            print ("Вы проиграли. Было загадано:", n, sep = ' ')
            break
        elif user_change < n: print("Загаданное число больше")
        elif user_change > n: print("Загаданное число меньше")
        user_try += 1
    print("Хотите начать сначала? (1 - ДА )")
    user_change = input()
    if user_change == "1": u = 1
    else: u = 0

File: Practice/15/C++/ex15/ex15/ex15.cpp
#include <iostream>
#include <ctime>

int main() {
	setlocale(LC_ALL, "ru");
	int u = 1;
	std::cout << "Здравствуйте! Вас приветствует игра \"Угадай число\". Компьютер загадает случайное число от 0 до 100.\nВаша задача угадать это число. На всё про всё вам даётся 5 попыток. Желаем вам удачи!" << std::endl;
	while(u) {
		std::cout << "Поехали!" << std::endl;
		srand(time(0));
		int userChange; int maxUserTry = 5;
		int n = 0 + rand() % 100 + 1;
		for (int userTry = 1; userTry <= maxUserTry; userTry++) {
			std::cin >> userChange;
			if (userChange == n) {
				std::cout << "Поздравляю! Вы угадали" << std::endl;
				break;
			}
			else if (userTry == maxUserTry) {
				std::cout << "Вы проиграли. Было загадано: " << n << std::endl;
				break;
			}
			else if (userChange < n) std::cout << "Загаданное число больше" << std::endl;
			else if (userChange > n) std::cout << "Загаданное число меньше" << std::endl;
		}
		std::cout << "Хотите начать сначала? (1 - ДА )" << std::endl;
		std::cin >> userChange;
		if (userChange == 1) u = 1;
		else u = 0;
	}	
}